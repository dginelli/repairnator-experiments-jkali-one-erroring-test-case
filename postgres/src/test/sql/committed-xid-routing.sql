CREATE SCHEMA e_0010;
CREATE TABLE e_0010.tas_tbl
(
  pk_tas bigserial NOT NULL,
  tas_name text,
  stuff json,
  CONSTRAINT idx_e_0010_tas_tbll_pkey PRIMARY KEY (pk_tas)
);

CREATE INDEX es_e_0010_tas_tbl ON e_0010.tas_tbl USING zombodb (zdb('e_0010.tas_tbl', ctid), zdb(tas_tbl.*))
WITH (url='http://localhost:9200/', replicas='0', shards='32');


INSERT INTO e_0010.tas_tbl(tas_name, stuff) VALUES('john doe', '[{"color":"blue"}]');
INSERT INTO e_0010.tas_tbl(tas_name, stuff) VALUES('jane doe', '[{"color":"pink"}]');
INSERT INTO e_0010.tas_tbl(tas_name, stuff) VALUES('bob smith', '[{"color":"blue"},{"color":"green"}]');


SELECT * FROM e_0010.tas_tbl;

SELECT * FROM e_0010.tas_tbl, json_array_elements(stuff) as elem WHERE elem->>'color' = 'blue';
SELECT * FROM zdb_estimate_count('e_0010.tas_tbl', 'stuff.color:"blue"');


UPDATE e_0010.tas_tbl
SET stuff = NULL
WHERE pk_tas = 3;


SELECT * FROM zdb_estimate_count('e_0010.tas_tbl', 'stuff.color:"blue"');

DROP SCHEMA e_0010 CASCADE ;