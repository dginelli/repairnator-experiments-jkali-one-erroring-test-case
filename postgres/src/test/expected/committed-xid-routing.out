CREATE SCHEMA e_0010;
CREATE TABLE e_0010.tas_tbl
(
  pk_tas bigserial NOT NULL,
  tas_name text,
  stuff json,
  CONSTRAINT idx_e_0010_tas_tbll_pkey PRIMARY KEY (pk_tas)
);
CREATE INDEX es_e_0010_tas_tbl ON e_0010.tas_tbl USING zombodb (zdb('e_0010.tas_tbl', ctid), zdb(tas_tbl.*))
WITH (url='http://localhost:9200/', replicas='0', shards='32');
INSERT INTO e_0010.tas_tbl(tas_name, stuff) VALUES('john doe', '[{"color":"blue"}]');
INSERT INTO e_0010.tas_tbl(tas_name, stuff) VALUES('jane doe', '[{"color":"pink"}]');
INSERT INTO e_0010.tas_tbl(tas_name, stuff) VALUES('bob smith', '[{"color":"blue"},{"color":"green"}]');
SELECT * FROM e_0010.tas_tbl;
 pk_tas | tas_name  |                stuff                 
--------+-----------+--------------------------------------
      1 | john doe  | [{"color":"blue"}]
      2 | jane doe  | [{"color":"pink"}]
      3 | bob smith | [{"color":"blue"},{"color":"green"}]
(3 rows)

SELECT * FROM e_0010.tas_tbl, json_array_elements(stuff) as elem WHERE elem->>'color' = 'blue';
 pk_tas | tas_name  |                stuff                 |      value       
--------+-----------+--------------------------------------+------------------
      1 | john doe  | [{"color":"blue"}]                   | {"color":"blue"}
      3 | bob smith | [{"color":"blue"},{"color":"green"}] | {"color":"blue"}
(2 rows)

SELECT * FROM zdb_estimate_count('e_0010.tas_tbl', 'stuff.color:"blue"');
 zdb_estimate_count 
--------------------
                  2
(1 row)

UPDATE e_0010.tas_tbl
SET stuff = NULL
WHERE pk_tas = 3;
SELECT * FROM zdb_estimate_count('e_0010.tas_tbl', 'stuff.color:"blue"');
 zdb_estimate_count 
--------------------
                  1
(1 row)

DROP SCHEMA e_0010 CASCADE ;
NOTICE:  drop cascades to table e_0010.tas_tbl
